### ISSUE TITLE
VoilaExecutePreprocessor.allow_errors always True

### ISSUE THREAD

**Issue by jeffyjefflabs** [voila-dashboards/voila#519]
If a cell throws an exception, the cells after it are still being executed.  Setting allow_errors explicitly to False (which is the default anyway) doesn't affect it.

**Comment by jtpio** [comment]
Good catch, thanks @jeffyjefflabs.

Voila handles `CellExecutionError` here: https://github.com/voila-dashboards/voila/blob/919d57db3a143566009853f27ca05131c4b23da7/voila/execute.py#L148

Which seems to be shadowing the `allow_errors` flag from nbconvert.

**Comment by martinRenou** [comment]
I am working on this today

**Comment by martinRenou** [comment]
It comes from https://github.com/voila-dashboards/voila/pull/186, I feel like this PR should not have been merged as it shadows the `CellExecutionError` generated by nbconvert when `allow_errors` is set to False (see https://github.com/jupyter/nbconvert/blob/master/nbconvert/preprocessors/execute.py#L477)

**Comment by martinRenou** [comment]
Although I agree with the initial intention of this PR. Maybe we should set `allow_errors` to True by default in Voila?

**Comment by maartenbreddels** [comment]
I agree, since this looks to be controllable from nbconvert, voila should not add similar behavior. Maybe allow_error should be set to True on debug?

**Comment by martinRenou** [comment]
What I have in mind is to overwrite the `allow_errors` default value to `True` in Voila, so that Voila does not show a 500 error by default if the execution failed.

And in the case of having `allow_errors` to True, errors are only shown in debug mode.

**Comment by maartenbreddels** [comment]
Sound good I think, maybe we can use this pattern:
https://github.com/voila-dashboards/voila/blob/919d57db3a143566009853f27ca05131c4b23da7/voila/exporter.py#L62

**Comment by martinRenou** [comment]
Completely overwriting the trait would allow writing the right documentation in `voila --help-all`:
```Python
allow_errors = Bool(
        True,
        help=dedent(
            """
            If `True` (default), execution errors are ignored and the execution
            is continued until the end of the notebook. Output from
            exceptions is included in the cell output only if Voila is run with --debug.
            If `False`, when a cell raises an error the
            execution is stopped and a `CellExecutionError`
            is raised.
            """
        )
    ).tag(config=True)
```

**Comment by maartenbreddels** [comment]
I remember this being an issue actually, not 100% sure it is related: https://github.com/jupyter/nbconvert/pull/1056#issuecomment-516834845

**Comment by martinRenou** [comment]
I am not sure I understand the comment on your link, what was the issue?

**Comment by maartenbreddels** [comment]
If you add --allow-errors to aliases (like in https://github.com/jupyter/nbconvert/commit/e78f02eeb77a1cf5eb871dae7ebf18eee15a6c72) traitlets gets confused which class to alias this to. But as long as we don't make it an alias we should be fine.

**Comment by jeffyjefflabs** [comment]
Thanks for looking into this!  I tried your PR #530 and it works for me.  I like keeping `allow_errors=False` by default; for most of our use cases we definitely don't want to try to keep going after an error.
